pdns/dnsdist-lua.cc

```diff
@@ -1987,6 +1988,19 @@ static void setupLuaConfig(LuaContext& luaCtx, bool client, bool configCheck)
   });
 
   luaCtx.writeFunction("setTCPInternalPipeBufferSize", [](uint64_t size) { g_tcpInternalPipeBufferSize = size; });
+  luaCtx.writeFunction("setTCPFastOpenKey", [](const std::string& keyString) {
+    setLuaSideEffect();
+    unsigned int key[4] = {};
+    auto ret = sscanf(keyString.c_str(), "%x-%x-%x-%x", &key[0], &key[1], &key[2], &key[3]);
```

Comment:
This way of representing and reading the key is endian dependent. I'd go for a Base64 encoding.

Reply:
I see now this is the repsentation Linux uses for their `sysctl. So using this reperesentation has it's advantages. It should be tested though if setting via `dnsdist` and setting via the `sysctl` produces the same fast open cookies.

Reply:
Yes, this method relies on endianness. I also considered using `Base64` to solve it. But I wrote this because of the [`self test`](https://github.com/torvalds/linux/blob/1329e40ebd18f1171f89f9e61b7a4aed9a1de406/tools/testing/selftests/net/tcp_fastopen_backup_key.c#L62) of the linux kernel.

Reply:
Hi, I wrote a simple program to test if setting `TCP_FASTOPEN_KEY` like this is consistent with setting it using `sysctl`. I open two ports, one of which uses the method in the pull request to modify `TCP_FASTOPEN_KEY` and the other does not. 

By comparing the FastOpen Cookie changes before and after `sysctl`, I think my modification will be consistent with `sysctl` (at least in `x86_64` environment).

test environment:
> ArchLinux 5.17.1-arch1-1 x86_64

Test Program and Test Result: https://gist.github.com/Y7n05h/8bc5873b8e80951dd545de86fbd9f388